packages/jetbrains-plugin/src/test/kotlin/com/mongodb/jbplugin/fixtures/components/MdbJavaEditorToolbarFixture.kt

```diff
@@ -43,11 +45,36 @@ class MdbJavaEditorToolbarFixture(
                 timeout = 50.milliseconds.toJavaDuration()
             )
         }.isSuccess
-    fun detachDataSource() {
-        eventually {
-            dataSources.selectItem("Detach data source")
-            if (!dataSources.selectedText().contains("Attach MongoDB")) {
-                throw Exception("Could not detach data source")
+
+    fun selectDataSource(title: String) {
+        eventually(1.minutes.toJavaDuration()) {
+            step("Selecting DataSource $title in toolbar") {
```

Comment:
```suggestion
            step("Selecting DataSource $title in toolbar") {
                eventually(1.minutes.toJavaDuration()) {
```

Because eventually will retry every a few milliseconds, until (in this case) 1 minute passes, so we will see the step in the logs for every retry.

Reply:
If we are going to use this pattern of eventually/step a lot, maybe we want to have an eventually that receives the step name and does both things? So we encapsulate both calls and it's easier to use.

Reply:
Actually in practice I found that step log for every retries were making it more clear about what is happening. I am thinking of step as very detailed and verbose explanation of what is happening in the UI. Agree that we probably should have eventually with a step block. I will work towards that if my earlier reasoning sounds good?

Reply:
It sounds great, if it's possible can we add then the retry number? Like "Selecting DataSource $title: retry N" so it's clear in the logs that we are retrying?

Reply:
Added this [here](https://github.com/mongodb-js/intellij/pull/78#discussion_r1812621034)